### PR TITLE
Implement PlatformLocale and StringDelegate for k/wasm

### PR DESCRIPTION
**Pull request by eymar**
The implementation relies on JS browser API

NOTE: the PR is to wasm-main branch!

### CODE REVIEW

File: compose/ui/ui-text/src/jsWasmMain/kotlin/androidx/compose/ui/text/intl/JsPlatformLocale.js.kt [hunk 1]
```diff
@@ -16,29 +16,51 @@
...
-    override fun toLanguageTag(): String = TODO("implement native toLanguageTag") // locale.toLanguageTag()
+    override fun toLanguageTag(): String = locale.baseName
 }
 
 internal actual fun createPlatformLocaleDelegate(): PlatformLocaleDelegate =
     object : PlatformLocaleDelegate {
         override val current: LocaleList
-            get() = LocaleList(listOf(Locale(JsLocale(Any()))))
+            get() = LocaleList(
+                listOf(Locale(JsLocale(currentLanguageTag())))
```
Comment (MatkovIvan) [inline compose/ui/ui-text/src/jsWasmMain/kotlin/androidx/compose/ui/text/intl/JsPlatformLocale.js.kt]: We're dealing with list here, so let's use [`navigator.languages`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/languages) instead

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

File: compose/ui/ui-text/src/jsWasmMain/kotlin/androidx/compose/ui/text/intl/JsPlatformLocale.js.kt (hunk 1 above)
Comment (eymar) [reply compose/ui/ui-text/src/jsWasmMain/kotlin/androidx/compose/ui/text/intl/JsPlatformLocale.js.kt]: fixed

File: compose/ui/ui-text/src/jsWasmMain/kotlin/androidx/compose/ui/text/intl/JsPlatformLocale.js.kt [hunk 2]
```diff
@@ -16,29 +16,56 @@
...
+internal external interface IntlLocale {
+    val language: String
+    val script: String?
+    val region: String?
+    val baseName: String
+}
+
+internal fun parseLanguageTagToIntlLocale(languageTag: String): IntlLocale = js("new Intl.Locale(languageTag)")
+
+internal fun currentLanguageTag(): String = js("window.navigator.language")
```
Comment (MatkovIvan) [inline compose/ui/ui-text/src/jsWasmMain/kotlin/androidx/compose/ui/text/intl/JsPlatformLocale.js.kt]: It's unused now

Comment (eymar) [reply compose/ui/ui-text/src/jsWasmMain/kotlin/androidx/compose/ui/text/intl/JsPlatformLocale.js.kt]: ack. removed it

**Review by MatkovIvan — APPROVED** [review]

**Pull request merged**